ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -892,6 +911,50 @@ void trackAgeAtAccess(long creationTimeMs) {
       }
     }
   }
+
+  /**
+   * Close {@link NonBlockingRouterMetrics} by shutting down scheduler (if present) in this class.
+   */
+  public void close() {
+    if (scheduler != null) {
+      shutDownExecutorService(scheduler, 5, TimeUnit.SECONDS);
+    }
+  }
+
+  /**
+   * A thread that helps periodically dump resource-level histogram (with given percentile) into log file.
+   */
+  private class HistogramDumper implements Runnable {
+
+    @Override
+    public void run() {
+      double quantile = routerConfig.routerLatencyToleranceQuantile;
+      for (Map.Entry<Resource, Histogram> resourceToHistogram : getBlobLocalDcResourceToLatency.entrySet()) {
+        Resource resource = resourceToHistogram.getKey();
+        Histogram histogram = resourceToHistogram.getValue();
+        logger.info("{} GetBlob local DC latency histogram {}th percentile in ms: {}", resource.toString(),
+            quantile * 100, histogram.getSnapshot().getValue(quantile));
+      }
+      for (Map.Entry<Resource, Histogram> resourceToHistogram : getBlobCrossDcResourceToLatency.entrySet()) {
+        Resource resource = resourceToHistogram.getKey();
+        Histogram histogram = resourceToHistogram.getValue();
+        logger.info("{} GetBlob cross DC latency histogram {}th percentile in ms: {}", resource.toString(),
+            quantile * 100, histogram.getSnapshot().getValue(quantile));
+      }
+      for (Map.Entry<Resource, Histogram> resourceToHistogram : getBlobInfoLocalDcResourceToLatency.entrySet()) {
+        Resource resource = resourceToHistogram.getKey();
```

Comment:
Create a function, method or class to make a general histgram dumper?

Reply:
Sure but I need to think about this. Will make the change in future PR.